utils/mgmt_system/google.py

```diff
@@ -0,0 +1,269 @@
+            # internet.
+            'networkInterfaces': [{
+                'network': 'global/networks/default',
+                'accessConfigs': [
+                    {'type': 'ONE_TO_ONE_NAT', 'name': 'External NAT'}
+                ]
+            }],
+
+            # Allow the instance to access cloud storage and logging.
+            'serviceAccounts': [{
+                'email': 'default',
+                'scopes': [
+                    'https://www.googleapis.com/auth/devstorage.read_write',
+                    'https://www.googleapis.com/auth/logging.write'
+                ]
+            }],
+
+            # Metadata is readable from the instance and allows you to
+            # pass configuration from deployment scripts to instances.
+            'metadata': {
+                'items': [{
+                    # Startup script is automatically executed by the
+                    # instance upon startup.
+                    'key': 'startup-script',
+                    'value': script
+                }, {
+                    # Every project has a default Cloud Storage bucket that's
+                    # the same name as the project.
+                    'key': 'bucket',
+                    'value': self._project
+                }]
+            }
+        }
+
+        operation = self._compute.instances().insert(project=self._project,
+                                                     zone=self._zone,
+                                                     body=config).execute()
+        self.wait_vm_running(operation['name'])
+        return True
+
+    def delete_vm(self, instance_name):
+        logger.info(" Deleting Google Cloud instance %s" % instance_name)
+        operation = self._compute.instances().delete(project=self._project,
+                                         zone=self._zone, instance=instance_name).execute()
+        self.wait_vm_running(operation['name'])
+        return True
+
+    def restart_vm(self, instance_name):
+        logger.info(" Restarting Google Cloud instance %s" % instance_name)
+        operation = self._compute.instances().reset(project=self._project,
+                                       zone=self._zone, instance=instance_name).execute()
+        self.wait_vm_running(operation['name'])
+        return True
+
+    def stop_vm(self, instance_name):
+        logger.info(" Stoping Google Cloud instance %s" % instance_name)
+        operation = self._compute.instances().start(project=self._project,
+                                        zone=self._zone, instance=instance_name).execute()
+
+        self.wait_vm_running(operation['name'])
+        return True
+
+    def start_vm(self, instance_name):
+        # This method starts an instance that was stopped using the using the
+        # instances().stop method.
+        logger.info(" Starting Google Cloud instance %s" % instance_name)
+        operation = self._compute.instances().start(project=self._project,
```

Comment:
Does this need to catch an Exception if you try to start and already started Instance? Do we need to check the state before we issue the start?